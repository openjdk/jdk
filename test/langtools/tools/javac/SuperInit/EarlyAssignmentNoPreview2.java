/*
 * @test /nodynamiccopyright/
 * @bug 8334258
 * @summary Disallow early assignment if FLEXIBLE_CONSTRUCTORS preview feature is not enabled
 * @compile/fail/ref=EarlyAssignmentNoPreview2.out -XDrawDiagnostics EarlyAssignmentNoPreview2.java
 */
public class EarlyAssignmentNoPreview2 {

    Runnable r;

    public EarlyAssignmentNoPreview2() {
        this(this.r = () -> System.out.println("hello"));
    }

    public EarlyAssignmentNoPreview2(Runnable r) {
    }

    public static void main(String[] args) {
        new EarlyAssignmentNoPreview2();
    }
}
