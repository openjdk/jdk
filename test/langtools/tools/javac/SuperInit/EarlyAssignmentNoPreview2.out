EarlyAssignmentNoPreview2.java:12:14: compiler.err.feature.not.supported.in.source: (compiler.misc.feature.flexible.constructors), 24, 25
1 error
